File: lib/models/Notification.ts
import mongoose, { Schema, model, models } from 'mongoose';

const notificationSchema = new Schema({
  userId: { type: String, required: true, index: true },
  type: {
    type: String,
    enum: ['calibration_due', 'cert_expiring', 'new_request', 'request_update', 'system', 'admin'],
    required: true
  },
  title: { type: String, required: true },
  message: { type: String, required: true },
  read: { type: Boolean, default: false },
  link: { type: String, default: null },
  createdAt: { type: Date, default: Date.now },
});

export const Notification = models?.Notification || model('Notification', notificationSchema);

File: mobile/lib/features/notifications/models/notification.dart
// lib/features/notifications/models/notification.dart

class AppNotification {
  const AppNotification({
    required this.id,
    required this.title,
    required this.message,
    required this.read,
    required this.createdAt,
    this.type,
    this.link,
  });

  final String id;
  final String title;
  final String message;
  final bool read;
  final DateTime createdAt;
  final String? type;
  final String? link;

  factory AppNotification.fromJson(Map<String, dynamic> json) {
    DateTime parseDate(dynamic v) {
      if (v == null) return DateTime.now();
      try {
        return DateTime.parse(v.toString()).toLocal();
      } catch (_) {
        return DateTime.now();
      }
    }

    return AppNotification(
      id: json['id']?.toString() ?? json['_id']?.toString() ?? '',
      title: json['title']?.toString() ?? json['type']?.toString() ?? 'Notification',
      message: json['message']?.toString() ?? '',
      read: json['read'] == true,
      createdAt: parseDate(json['createdAt']),
      type: json['type']?.toString(),
      link: json['link']?.toString(),
    );
  }
}
